docs/style/CodingGuidelines.md

```diff
@@ -0,0 +1,89 @@
+
+If your team shares development PCs, please note that uncommitted changes on those public PCs or Laptops are deleted w/o further research. Thus, keep them clean.
+
+## Visuals
+
+* Spaces instead of Tabs. Use four spaces for normal indentation.
+* Do not indent namespaces. Almost all code is inside a namespace or even nested namespace. Indentation here wastes precious screen real estate.
+* If possible use the common clang-format file that resides aside of these Coding Guidelines and integrate it into your workflow/IDE.
+
+## General Remarks
+
+* Write C++11 not C and avoid utilising Boost when this needed feature is in C++11, already.
+* Try to write Object-Oriented code.
+* Avoid code duplication at all costs.
+
+## Naming Conventions
+
+* Classnames and structs are written in `CamelCase` (first letter capitalized).
+* Method names are written in `camelCase` (first letter lower case).
+* Member variables are written in `_camelCase`, i.e., lower case first letter and prefixed with an underscore.
+
+## Type Operators and Modifiers
+
+* The usage of  `const` is encouraged. `const` increases readability and safety of the codebase.
+* Pointer and Reference Expressions:
+When declaring pointers and references, prefer placing the operator adjacent to the type rather than the name. This improves consistency when using the type in other expressions, such as templates or casts.
+   * Prefer `int* p` instead of `int *p`.
+   * Prefer `Vector3& pos` over `Vector3 &pos`.
+   
+## Includes
+
+* in header files do forward declaration instead of includes whenever possible: It compiles faster and avoids cycling include dependencies.
+* use `"Bla.h"` for header files that are inside your local include folder
+* use `<Bla.h>` for header files that are outside your local include folder
+* sort the header files from local to general:
+    1. include your own header: `#include "MyClass.h"`
+    2. include header from your include folder: `#include "UDPConnection.h"`
+    3. include from our own projects: `#include <engine/model/Plan.h>`
+    4. include from third party libraries: `#include <ros/ros.h>`
+    5. include system library headers: `#include <vector>`
+* sort the includes alphabetically in each section and keep a newline between those sections (this is done automatically by clang-format)
+* don't include with relative paths, e.g., `#include "../../folder/header.h"`
+
+## Namespaces
+
+- always use at least one namespace
+- As a rule of thumb - Use one namespace per repository:
+    * supplementary: supplementary[::nested_namespace]
+    * alica: alica[::nested_namespace]
+
+## Header
+
+* use `#pragma once` instead of include guards whenever possible: Nowadays all common compilers support that feature and it facilitates refactoring and avoids include guard clashes.
+* have at most one 'public:', 'protected:', and 'private:' section and list them in this order
+* in each section place functions first, then member variables
```

Comment:
I might add here: Order the member variables within a struct or class by decreasing usage. This isn't important if the entire object is less than 64 bytes - a cache line - but it will help in cases where the object is larger and commonly referenced data is not in the first cache line.

Reply:
While that is good advice, it is not the only consideration for member ordering.
Padding between differently aligned types also plays a role (which yields the rule of thumb: biggest member first).
I'm on the fence, but I think I'd rather keep this out of the coding style, as it is more a coding topic that needs to be considered on a case by case basis.